integration-cli/docker_cli_build_test.go

```diff
@@ -4941,6 +4941,40 @@ func (s *DockerSuite) TestBuildLabelsCache(c *check.C) {
 
 }
 
+func (s *DockerSuite) TestBuildNotVerbose(c *check.C) {
+	// This test makes sure that the `-s` mode works correctly
+	testRequires(c, DaemonIsLinux)
+	tt := []struct {
+		TestName     string // Silent Build Name
+		BuildCmds    string // Build string
+		ShouldFail   bool   // Expect to fail
+		StdoutRegexp string
+		StderrRegexp string
+	}{
+		{"testsilentbuild_busybox_and_echo", "FROM busybox\nRUN echo hello", false, "^(sha256:|)[a-z0-9]{64}\\n$", "^$"},
+		{"testsilentbuild_invalid_image", "FROM blabla", true, "^$", "Step 1 : FROM blabla((\\n|\r)+.*)+Error: image library/blabla(:latest)? not found\\n"},
+		{"testsilentbuild_exit_code_1", "FROM busybox\nRUN echo hello\nRUN exit 1", true, "^$", "Step 1 : FROM busybox((\\n|\r)+.*)+RUN echo hello((\\n|\r)+.*)+hello((\\n|\r)+.*)+Step 3 : RUN exit 1((\\n|\r)+.*)+The command '/bin/sh -c exit 1' returned a non-zero code: 1"},
```

Comment:
Instead of matching patterns in the output of `DisplayJSONMessagesStream`, I would make an alternative implementation that coalesces messages based on the `ID` field of `JSONMessage`.